swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/Store.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  Store.swift
+//  (c) 2024 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import Combine
+import NetworkExtension
+import UserNotifications
+import OSLog
+
+#if os(macOS)
+  import AppKit
+#endif
+
+@MainActor
+public final class Store: ObservableObject {
+  @Published private(set) var actorName: String?
+
+  // Make our tunnel configuration convenient for SettingsView to consume
+  @Published private(set) var settings: Settings
+
+  // Enacapsulate Tunnel status here to make it easier for other components
+  // to observe
+  @Published private(set) var status: NEVPNStatus
+
+  // This is not currently updated after it is initialized, but
+  // we could periodically update it if we need to.
+  @Published private(set) var decision: UNAuthorizationStatus
+
+  private let tunnelManager: TunnelManager
+  private var sessionNotification: SessionNotification
+  private var cancellables: Set<AnyCancellable> = []
+  private var resourcesTimer: Timer?
+
+  public init() {
+    self.status = .disconnected
+    self.decision = .authorized
+    self.settings = Settings.defaultValue
+
+    self.tunnelManager = TunnelManager()
+    self.sessionNotification = SessionNotification()
+
+    initNotifications()
+    initTunnelManager()
+  }
+
+  private func initNotifications() {
+    // Finish initializing notification binding
+    sessionNotification.signInHandler = {
+      Task { await WebAuthSession.signIn(store: self) }
+    }
+
+    sessionNotification.$decision
+      .receive(on: DispatchQueue.main)
+      .sink(receiveValue: { [weak self] decision in
+        guard let self = self else { return }
+        self.decision = decision
+      })
+      .store(in: &cancellables)
+  }
+
+  private func initTunnelManager() {
+    // Subscribe to status updates from the tunnelManager
+    tunnelManager.statusChangeHandler = handleVPNStatusChange
+
+    // Load our existing VPN profile and initialize our state
+    tunnelManager.load() { loadedStatus, loadedSettings, loadedActorName in
+      DispatchQueue.main.async {
+        self.status = loadedStatus
+
+        if let loadedSettings = loadedSettings {
+          self.settings = loadedSettings
+        }
+
+        if let loadedActorName = loadedActorName {
+          self.actorName = loadedActorName
+        }
```

Comment:
So if the tunnel doesn't send us one of these, we keep whatever we had before, right? 

Just doing this wouldn't be the right behavior?
```
self.settings = loadedSettings
self.actorName = loadedActorName
```

Reply:
No because these are observed properties, so setting them will update whatever's watching them. In the case of Settings, it will set them to `nil` when we start without a tunnel as opposed to leaving them as `Settings.defaultValue` set above in the initializer.